frontend/src/libs/components/input-file/input-file.tsx

```diff
@@ -0,0 +1,90 @@
+import { FIRST_ARRAY_INDEX } from '#libs/constants/constants.js';
+import { IconColor } from '#libs/enums/enums.js';
+import {
+  getDeepErrorMessage,
+  getValidClassNames,
+} from '#libs/helpers/helpers.js';
+import { useCallback, useFormController } from '#libs/hooks/hooks.js';
+import {
+  type FormControl,
+  type FormFieldErrors,
+  type FormFieldPath,
+  type FormFieldValues,
+} from '#libs/types/types.js';
+
+import { Icon } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties<T extends FormFieldValues> = {
+  control: FormControl<T, null>;
+  errors: FormFieldErrors<T>;
+  name: FormFieldPath<T>;
+  fileName: string | null;
+  label: string;
+  description: string;
+  onChange?: (file: File) => void;
+};
+
+const InputFile = <T extends FormFieldValues>({
+  control,
+  errors,
+  name,
+  fileName,
+  label,
+  description,
+  onChange,
+}: Properties<T>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+
+  const errorMessage = getDeepErrorMessage(errors[name]);
```

Comment:
let's remove this helper and access errors similar to this:
```suggestion
  const sizeError = errors[`name.${size}`];
  const otherError = errors[`name.${otherProperty}`];
```
this is to rare case to create a common helper for this and we can simply handle this logic without it

Reply:
@Liza-Veis Maybe I'm missing smth, but the way you suggested doesn't work and returns `undefined`. I also tried the way shown on screenshot, but it requires lots of messy type assertion

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92298163/cf596faa-b397-4312-9c34-e38d7a16774a)

![image](https://github.com/BinaryStudioAcademy/bsa-2023-calmpal/assets/92298163/dca647e6-22b3-4992-a950-78ce5f4aa319)